### PR TITLE
added color for each legendEntry(if not set, using default legend color)

### PR DESCRIPTION
**Pull request by SeRG1k17**
Simple way add color for each legendEntry.
Usage:

```
    NSMutableArray *legendEntries = [[NSMutableArray alloc] init];
    
    for (int i = 0; i < count; i++)
    {
        UIColor *entryColor = colors[i];
        ChartLegendEntry *entry = [[ChartLegendEntry alloc] initWithLabel:parties[i % parties.count]
                                                               labelColor:entryColor // <-- New property
                                                                     form:ChartLegendFormCircle
                                                                 formSize:8
                                                            formLineWidth:0
                                                        formLineDashPhase:0
                                                      formLineDashLengths:0
                                                                formColor:entryColor];
        
        [legendEntries addObject:entry];
    }
    
    [_chartView.legend setCustomWithEntries: legendEntries];
```

### CODE REVIEW

File: Source/Charts/Renderers/LegendRenderer.swift [hunk 1]
```diff
@@ -431,6 +431,10 @@ open class LegendRenderer: Renderer
                 
                 var posX = originPosX
                 
+                if let labelColor = e.labelColor {
```
Comment (jjatie) [inline Source/Charts/Renderers/LegendRenderer.swift]: `{` should be on a new line.

File: Source/Charts/Renderers/LegendRenderer.swift [hunk 2]
```diff
@@ -205,7 +205,7 @@ open class LegendRenderer: Renderer
         }
         
         let labelFont = legend.font
-        let labelTextColor = legend.textColor
+        var labelTextColor = legend.textColor
```
Comment (jjatie) [inline Source/Charts/Renderers/LegendRenderer.swift]: this should be `let legendTextColor...`

File: Source/Charts/Renderers/LegendRenderer.swift (hunk 1 above)
Comment (jjatie) [inline Source/Charts/Renderers/LegendRenderer.swift]: this should be `let labelTextColor = e.labelColor ?? legendTextColor`

Comment (SeRG1k17) [reply Source/Charts/Renderers/LegendRenderer.swift]: @jjatie It's been a long time. I have to double-check it

File: Source/Charts/Renderers/LegendRenderer.swift (hunk 2 above)
Comment (SeRG1k17) [reply Source/Charts/Renderers/LegendRenderer.swift]: @jjatie It's been a long time. I have to double-check it

**Comment by petester42** [comment]
closed in favour of #3558

**Pull request closed without merging**